### PR TITLE
Arm on Docker

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
* Create snippet for arm for Docker
* Add snippets to Arm page and config reference and building Docker images pages
* Update Using Arm page to specify that is Arm VM and add section linking to Arm on Docker as an alternative

To give an idea of how it looks:

Screenshot 2023-07-20 at 23 04 32
Screenshot 2023-07-20 at 23 04 44

# Reasons
New feature

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Comment by sebastian-lerner** [comment]
looks good, thank you! do you want to link to the Discuss post anywhere? to talk about the fact that not all of the convenience images are out quite yet.  i would also add a note that final specs are still TBD (they're likely gonna change a bit).

and then last comment: I mention an important point at the bottom of the[ discusss post](https://discuss.circleci.com/t/product-launch-arm-docker-preview/48601) that we should probably add to the docs: The [Docker version for Remote Docker 9](https://circleci.com/docs/building-docker-images/#docker-version) with ARM will only support `default` & `edge`.

So where we talk about all of the different remote docker versions, we may want to  be explicit that arm only supports a subset of the ones listed.

**Review by sebastian-lerner — COMMENTED** [review]
left comments

File: jekyll/_cci2/building-docker-images.md [hunk 1]
```diff
@@ -74,10 +75,18 @@ jobs:
...
+* [Docker Layer Caching](/docs/glossary/#docker-layer-caching) (DLC) is enabled to speed up image building. (line 14)
 * We use [project environment variables](/docs/set-environment-variable/#set-an-environment-variable-in-a-project) to store credentials for Docker Hub. (line 19)
 
+### Resource classes
+{: #resource-classes}
+
+**Arm on Docker** Support for Arm architecture in the Docker execution environment is in **Preview**. For pricing information, and a list of CircleCI Docker convenience images that support Arm resource classes, see the [Arm and Docker Discuss post](https://discuss.circleci.com/t/product-launch-arm-docker-preview/48601).
+{: class="alert alert-caution"}
+
+The resource class specified for the primary container also applies to the remote Docker environment. Both x86 and Arm resource classes are available. For a full list of available resource classes see the [Configuration Reference](/docs/configuration-reference/#docker-execution-environment).
```
Comment (sebastian-lerner) [inline jekyll/_cci2/building-docker-images.md:88]: i thought about this some more and realized that this is not actually accurate.  for x86, if you specify a docker.xlarge & setup_remote_docker, the primary container & the VM are going to be 8 CPU / 32 GB RAM, not 8 CPU 16 GB RAM. https://circleci.com/product/features/resource-classes/

for ARM, we will use the ARM VM equivalent size when you use setup_remote_docker

**Review by sebastian-lerner — CHANGES_REQUESTED** [review]

**Review by kbatuigas — APPROVED** [review]
🚀

File: jekyll/_cci2/building-docker-images.md (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/building-docker-images.md:88]: Ahhh ok I see, I'll take another look at this section

**Review by sebastian-lerner — APPROVED** [review]
LGTM thank you

**Pull request merged**